sdk/rust/src/http.rs

```diff
@@ -461,14 +461,24 @@ impl IncomingResponse {
     /// # Panics
     ///
     /// Panics if the body was already consumed.
-    pub fn into_body_stream(self) -> impl futures::Stream<Item = Result<Vec<u8>, streams::Error>> {
+    // TODO: This should ideally take ownership of `self` and be called `into_body_stream` (i.e. symmetric with
+    // `IncomingRequest::into_body_stream`).  However, as of this writing, `wasmtime-wasi-http` is implemented in
+    // such a way that dropping an `IncomingResponse` will cause the request to be cancelled, meaning the caller
+    // won't necessarily have a chance to send the request body if they haven't started doing so yet (or, if they
+    // have started, they might not be able to finish before the connection is closed).  See
+    // https://github.com/bytecodealliance/wasmtime/issues/7413 for details.
+    pub fn take_body_stream(&self) -> impl futures::Stream<Item = Result<Vec<u8>, streams::Error>> {
```

Comment:
@dicej is this a planned change to the 2.0 SDK?

(Also, would this explain why your sample works and #1985 drops?)

Reply:
Yes and yes.  The API change is necessary because there's no other way to work around https://github.com/bytecodealliance/wasmtime/issues/7413, and I'm not sure when that will be fixed or whether it will be backported to Wasmtime 14 (and if it is, probably not in time for Spin 2.0).

Reply:
Update: here's a PR with the fix: https://github.com/bytecodealliance/wasmtime/pull/7426